connector/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisInputDStream.scala

```diff
@@ -17,9 +17,9 @@
 
 package org.apache.spark.streaming.kinesis
 
+import collection.JavaConverters._
```

Comment:
Can we just write it as scala.collection.*, and disallow other cases? Would be great to double check Scala guidance for this case.

Reply:
Is it feasible to add a `IllegalImportsChecker` item to `scalastyle-config.xml`  like:

```
  <check level="error" class="org.scalastyle.scalariform.IllegalImportsChecker" enabled="true">
    <parameters><parameter name="illegalImports"><![CDATA[collection.]]></parameter></parameters>
    <customMessage>Please use scala.collection instead.</customMessage>
  </check>
```

Reply:
1.I have checked the specifications of scala and there are no clear regulations.
2.At the same time, I searched the scala source code and found that in the `2.12.17` version,
the number of import writing methods in the form of `import scala.collection.JavaConverters._` is 37, while the number of import writing methods in the form of `import collection.JavaConverters._` is 3.

So, it is reasonable for us to prohibit importing in the form of `import collection.JavaConverters._` here.

3.In the `2.13.8` version, the number of import writing methods in the form of `import scala.collection.JavaConverters._` is 3, while the number of import writing methods in the form of `import collection.JavaConverters._` is 1.
According to official documents, in the 2.13 version, `collection.JavaConverters` has already been deprecated, so we need to consider this issue in the future.
<img width="789" alt="image" src="https://github.com/apache/spark/assets/15246973/1e759bbd-f066-4f96-92f0-db7875f6f084">
link: https://docs.scala-lang.org/overviews/core/collections-migration-213.html#deprecated-things-in-212-that-have-been-removed-in-213

Reply:
> Is it feasible to add a `IllegalImportsChecker` item to `scalastyle-config.xml` like:
> 
> ```
>   <check level="error" class="org.scalastyle.scalariform.IllegalImportsChecker" enabled="true">
>     <parameters><parameter name="illegalImports"><![CDATA[collection.]]></parameter></parameters>
>     <customMessage>Please use scala.collection instead.</customMessage>
>   </check>
> ```

Yes, that's right. Currently, this solution is more reasonable.